runtime/compiler/env/SharedCache.hpp

```diff
@@ -77,6 +78,10 @@ class TR_SharedCache
    // and also useful to represent "class chain not present" in places where a class chain offset is optional.
    static const uintptr_t INVALID_CLASS_CHAIN_OFFSET = 0;
 
+   // A uintptr_t value that can never represent a ROM method in an SCC. Used to check the return value of isMethodInSharedCache
+   // and related methods.
+   static const uintptr_t INVALID_ROM_METHOD_OFFSET = 1;
```

Comment:
ROM methods are stored as ROM structures, and their offsets are calculated here:

https://github.com/eclipse-openj9/openj9/blob/2902f0be5028883b1607c6d9d959ad83f76a399b/runtime/compiler/env/J9SharedCache.cpp#L778

using

https://github.com/eclipse-openj9/openj9/blob/2902f0be5028883b1607c6d9d959ad83f76a399b/runtime/compiler/env/J9SharedCache.hpp#L375

so 1 is guaranteed to be an invalid ROM method offset. Possibly 0 is also guaranteed to be invalid, but I'd have to look at the shared cache structure again. Maybe @dsouzai knows?

Reply:
I think that the offset cannot be 0 because ROMMethods are part of the ROMClass and ROMClasses have some other fields in the beginning, before the ROM methods.
Also, looking at this comment:
```
* \return An offset. Raises a fatal assertion before returning 0 if the pointer is invalid.
    */
   virtual uintptr_t offsetInSharedCacheFromROMMethod(J9ROMMethod *romMethod);
```
it seems that in the original code we already use an offset of 0 as invalid, though I agree that it will never take effect because of the fatal assert.

Reply:
All right. I can change the invalid offset to be 0 instead.

Reply:
Just saw your comment, Irwin. I can also add those sorts of constants as well.

Reply:
Though I think the names should be switched, because class chains are encoded from the end and rom structures are encoded from the start.

Reply:
Oh, I see. You do want them to denote valid encoded offsets from the start or end. I just assumed you meant for `OFFSET_FROM_*`  to denote an invalid offset according to that encoding directly.

Reply:
Ah, I forgot that `OFFSET_FROM_*` are already defined in `J9SharedCache`, so those aren't new constants.